### DISCUSSION TITLE
how to use different FS?

### DISCUSSION THREAD

**Discussion by qsmen** [occlum/occlum#909] [General]
Hi,
There are at least four file systems, that is the integrity only FS, encrypted FS, in-memory FS and host FS.   THe first two FS can be used for enclave image.  and the host FS can be used for communication between enclave and host.  my question is how to use the host FS?  the path with prefix "/host" is the only host FS?
 Thank you.

**Comment by liqinggd** [comment]
There are two ways to mount FS in Occlum

### Configure the “mount” part in Occlum.json.
When Occlum is started, it will read the Occlum.json and mount the FS automatically.
```
    // Mount points and their file systems
    //
    // The default configuration is shown below.
    "mount": [
        {
            "target": "/",
            "type": "unionfs",
            "options": {
                "layers": [
                    {
                        "target": "/",
                        "type": "sefs",
                        "source": "./build/mount/__ROOT",
                        "options": {
                            "MAC": ""
                        }
                    },
                    {
                        "target": "/",
                        "type": "sefs",
                        "source": "./run/mount/__ROOT"
                    }
                ]
            }
        },
        {
            "target": "/host",
            "type": "hostfs",
            "source": "."
        },
        {
            "target": "/proc",
            "type": "procfs"
        },
        {
            "target": "/dev",
            "type": "devfs"
        }
    ]
```
As the information is shown above, the host dir "." is mounted on "/host" of enclave as HostFS, it is easy to modify it.
```
        {
            "target": "/tmp",
            "type": "hostfs",
            "source": "/root/a"
        },
```
This configuration will mount the host dir "/root/a" on "/tmp" of enclave as HostFS.


### Use `mount` syscall to mount FS at runtime.
please see https://github.com/occlum/occlum/blob/master/docs/runtime_mount.md for the usage

**Reply by qsmen** [reply]
ok, got it. thank you sincerely.
